### ISSUE TITLE
set record in a macro in compose menu seems not to work properly

### ISSUE THREAD

**Issue by Knusper** [neomutt/neomutt#3491]
Here is something I can't wrap my head around.  I defined two  macros that easily let me switch between accounts for sending in the compose menu:
```
macro compose <f8> <edit-from>^U"James Bond <007@sis.gov.uk>"<enter>\
<enter-command>'source ~/.mutt/work_settings'<enter>
macro compose <f9> <edit-from>^U"James <james@mail.ru>"<enter>\
<enter-command>'source ~/.mutt/private_settings'<enter>
```

Now in `private_settings` and `work_settings` I have, amongst other things, `set record=+private/sent/` and `set record=+work/sent`, respectively.

If I run the the macros, everything gets  set up properly, especially the username, password and other SMTP settings.  Also `:set record` shows me that `record` was changed properly.  Still, when sending the mail, it is saved in the folder where `record` was pointing prior to running the macro. 😕 

Any ideas?

**Comment by gahr** [comment]
So you are saying that outgoing mails aren't saved according to `record`? Does that happen if you `:set record` manually? Do you have any folder-hook that might change `record` under the hood? Is `record` still correct after the mail is sent?

**Comment by Knusper** [comment]
Yes, I have folder hooks that change `record`.   But even if I change manually `:set record` in the compose menu, it does not have any effect.  So it has nothing to do with macros.

My folder hooks are like this:
```
folder-hook gov/* source ~/.mutt/work_settings
folder-hook mailru/* source ~/.mutt/private_settings
```

So, I am in some sub-folder of `gov/` and suddenly I decide to write to supervillains wife; of course she should not know that I am working for the government, so I change the from address, the smtp server, and also `:set record`, because I want to keep my emails tidy.  The mail is then sent correctly, with the correct from address and through the correct SMTP, but apparently after I sent the mail, the folder hook is executed prior the record is being written.   This seems to be an "interesting" feature of mutt. 😄   

>  Is record still correct after the mail is sent?

Well, its correct according to the folder where I am, but the value has changed according to the folder hook.

**Comment by gahr** [comment]
Uhm I can't replicate it (yet). What's the output of neomutt -v?

**Comment by Knusper** [comment]
@gahr  I am using a very outdated version of neomutt, namely the one that ships with debian 10 (NeoMutt 20180716).  However, if I look in the [ChangeLog.md](https://github.com/neomutt/neomutt/blob/main/ChangeLog.md), I see that a comment `call the folder-hook before saving to $record` in 2017.  I don't see any other obvious messages past that, which could indicate a change of this behaviour.

**Comment by gahr** [comment]
ohh alright.. sorry, I don't think anybody here is going to debug a version from 4 years ago... can you please try and reproduce with current master ?

**Comment by Knusper** [comment]
OK, I am going to try to compile the latest version soon.  Just one  more question:  After I send a mail from the compose menu (here bound to 'y'), how does neomutt determine for which folder the index menu is displayed afterwards -- is this configurable?

**Comment by gahr** [comment]
NeoMutt goes back to the mailbox the index was displaying before entering the compose screen.
